modin/core/execution/unidist/generic/modin_aqp.py

```diff
@@ -0,0 +1,119 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""The module for working with displaying progress bars for unidist engine."""
+
+import unidist
+import os
+import time
+import threading
+import warnings
+
+progress_bars = {}
+bar_lock = threading.Lock()
+
+
+def call_progress_bar(result_parts, line_no):
```

Comment:
https://github.com/modin-project/modin/issues/5275

Reply:
I wonder why do you want to introduce duplication in order to fix it right away, why not _move_ the code in this PR?..

Reply:
It would be better to me/us to merge the changes as soon as possible to start writing documentation, as well as fix the new created issues step by step. I do not want to bloat up this PR even more. My browser is already throttling when opening multiple comments to be resolved. Are you okay with that?